packages/core/src/lib/light_push/index.ts

```diff
@@ -50,7 +51,9 @@ class LightPush extends BaseProtocol implements ILightPush {
 
   constructor(libp2p: Libp2p, options?: ProtocolCreateOptions) {
     super(LightPushCodec, libp2p.components);
-    this.pubsubTopics = options?.pubsubTopics ?? [DefaultPubsubTopic];
+    this.pubsubTopics = options?.shardInfo
```

Comment:
We have a lot of these conditions now in the code. 
Let's create an abstraction for that and try to keep it in one place.